### PR TITLE
[MPS] Use MetalPerformancePrimitives for F.linear on macOS 26+

### PR DESCRIPTION
**Pull request by malfet**
Stack from [ghstack](https://github.com/ezyang/ghstack/tree/0.14.0) (oldest at bottom):
* #182038
* __->__ #182025

MPSNDArrayMatrixMultiplication and MPSGraph matrixMultiplication produce non-deterministic results for >2D fp16/bf16 inputs on M5 (macOS 26).
Replace them with a custom MPP matmul2d kernel on macOS 26+ that handles arbitrary strides (via contiguify+flatten), non-aligned sizes (via padding), and fuses 1D bias addition. Older macOS versions fall back to the existing nograph/MPSGraph paths unchanged.

Authored with Claude.

### CODE REVIEW

**Review by manuelcandales — APPROVED** [review]

File: aten/src/ATen/native/mps/kernels/Linear.metal [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    op.run(mA, mB, mC);
+  } else {
+    auto mA = mA_full.slice(0, (int)m_off);
+    auto mB = mB_full.slice(0, (int)n_off);
+    auto mC = mC_full.slice((int)n_off, (int)m_off);
+    op.run(mA, mB, mC);
+  }
+
+  if (HAS_BIAS) {
+    constexpr uint TG_SIZE = 4 * 32;
```
Comment (Isalia20) [inline aten/src/ATen/native/mps/kernels/Linear.metal:59]: I think barrier is needed here. When matmul2d::run finishes execution above, its device-memory writes to C won't be ordered against the += reads below.

i.e. this will result in non zero output:
```
import torch
import torch.nn.functional as F

dtype = torch.bfloat16

torch.manual_seed(0)
M, N, K = 26, 1024, 1024
x = torch.randn(M, K, dtype=dtype, device="mps")
w = torch.randn(N, K, dtype=dtype, device="mps")
b = torch.randn(N, dtype=dtype, device="mps")

fused = F.linear(x, w, b).cpu()
unfused = (F.linear(x, w) + b).cpu()
diff = (fused - unfused).abs()
print(f"max |fused - (matmul+bias)| = {diff.max()}")
```

max |fused - (matmul+bias)| = 107.5

**Review by Isalia20 — CHANGES_REQUESTED** [review]
